cfme/infrastructure/datastore.py

```diff
@@ -85,28 +66,23 @@ def delete(self, cancel=True):
         Note:
             Datastore must have 0 hosts and 0 VMs for this to work.
         """
-        sel.force_navigate('infrastructure_datastore', context=self._get_context())
+        self.load_details()
         cfg_btn('Remove from the VMDB', invokes_alert=True)
         sel.handle_alert(cancel=cancel)
 
     def wait_for_delete(self):
-        sel.force_navigate('infrastructure_datastores')
         wait_for(lambda: not self.exists, fail_condition=False,
              message="Wait datastore to disappear", num_sec=500, fail_func=sel.refresh)
 
     def wait_for_appear(self):
-        sel.force_navigate('infrastructure_datastores')
         wait_for(lambda: self.exists, fail_condition=False,
              message="Wait datastore to appear", num_sec=1000, fail_func=sel.refresh)
 
-    def load_details(self, refresh=False):
-        if not self._on_detail_page():
-            logger.debug("load_details: not on details already, navigating")
-            sel.force_navigate('infrastructure_datastore', context=self._get_context())
+    def load_details(self):
```

Comment:
I agree with you that it is needed.
However I'd propose to do this in another PR for all related objects as well as I did for cluster.